test/e2e/cruise/basic_test.go

```diff
@@ -230,6 +234,76 @@ var _ = Describe("Liqo E2E", func() {
 			}, generateTableEntries()...)
 		})
 
+		Context("E2E Storage Testing", func() {
+
+			const (
+				namespace = "storage-test"
+			)
+
+			var (
+				replica1Name = fmt.Sprintf("%v-1", storage.StatefulSetName)
+
+				changeSchedule = func(schedulable bool, nodeName string) {
+					virtualNodes := &corev1.NodeList{}
+					labels := client.MatchingLabels{liqoconst.TypeLabel: liqoconst.TypeNode}
+					if nodeName != "" {
+						labels[corev1.LabelHostname] = nodeName
+					}
+
+					Expect(testContext.Clusters[0].ControllerClient.List(ctx, virtualNodes, labels)).To(Succeed())
+					var err error
+					for i := range virtualNodes.Items {
+						virtualNodes.Items[i].Spec.Unschedulable = !schedulable
+						_, err = testContext.Clusters[0].NativeClient.CoreV1().Nodes().Update(ctx, &virtualNodes.Items[i], metav1.UpdateOptions{})
+						Expect(err).ToNot(HaveOccurred())
+					}
+				}
+
+				podIsPending = func(podName string) bool {
+					pod, err := testContext.Clusters[0].NativeClient.CoreV1().Pods(namespace).Get(ctx, podName, metav1.GetOptions{})
+					if err != nil {
+						return false
+					}
+					return pod.Status.Phase == corev1.PodPending
+				}
+			)
+
+			AfterEach(func() {
+				Expect(testContext.Clusters[0].NativeClient.CoreV1().Namespaces().Delete(ctx, namespace, metav1.DeleteOptions{})).To(Succeed())
+			})
+
+			It("run statefull app", func() {
+				By("Deploying the StatefulSet app")
+				err := storage.DeployApp(ctx, testContext.Clusters[0].Config, namespace)
+				Expect(err).ToNot(HaveOccurred())
+
+				By("Waiting until each pod of the application is ready")
+				options := k8s.NewKubectlOptions("", testContext.Clusters[0].KubeconfigPath, namespace)
+				storage.WaitDemoApp(GinkgoT(), options)
+
+				By("Checking that the pod is bound to a specific cluster")
+				pod, err := testContext.Clusters[0].NativeClient.CoreV1().Pods(namespace).Get(ctx, replica1Name, metav1.GetOptions{})
+				Expect(err).ToNot(HaveOccurred())
+				// cordon the virtual node
+				changeSchedule(false, pod.Spec.NodeName)
+
+				// delete the pod on the virtual node
```

Comment:
IMHO What about considering having this comment as By steps to simplify the debug?